Week3/homework/step4-bonus.js

```diff
@@ -3,8 +3,16 @@
 const values = ['a', 'b', 'c', 'd', 'a', 'e', 'f', 'c'];
 
 // Add your function here. Try and come up with a good name for this function
-
+function constructArrayWithUniqueValues(arr) {
+    const newValues = [];
+    for (const element of arr) {
```

Comment:
This is nice but would you like to try it with `Array.reduce`? 😉

Reply:
Otherwise, you can also try using `Set` in JavaScript. If you create a Set out of an Array, it is automatically going to trim out any duplicate values. However, you would have to create an Array back from the Set using the `Array.from` method. More information about `Set`s can be found here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

Reply:
const newValues = values.reduce((acc, value) => {
  if (acc.includes(value) === false) {
    acc.push(value);
  }
  return acc;
}, []);

console.log(newValues);